citar.el

```diff
@@ -1089,21 +1097,23 @@ are potentially cross-referenced from elements of KEYS."
 (defun citar-open (keys)
   "Open related resources (links or files) for KEYS."
   (interactive (list (citar-select-refs)))
-  (let ((resource-candidates
-         (citar--get-resource-candidates
-          keys :files t :notes t :links t)))
-    (cond
-     ((eq nil resource-candidates)
-      (error "No associated resources"))
-     ((unless citar-open-prompt
-        (eq 1 (length resource-candidates)))
-      (citar--open-multi (car resource-candidates)))
-     (t (citar--open-multi
-         (citar--select-resource keys :files t :notes t :links t))))))
+  (let ((embark-default-action-overrides `((file . ,#'citar-file-open)
+                                           (url . ,#'browse-url)
+                                           ,@(when-let ((notecat (citar--get-notes-config :category))
+                                                        (open (citar--get-notes-config :open)))
+                                               (list (cons notecat open)))
+                                           . ,embark-default-action-overrides)))
+      (if-let ((resource (citar--select-resource keys
+                                                 :files t :links t :notes t
+                                                 :always-prompt citar-open-prompt)))
+          ;; TODO Does this work for non-file notes?
```

Comment:
@roshanshariff - do you mean, for example, citar-org-roam?

If yes, the answer is yes; it works.

Reply:
I just tested it and `citar-open` doesn't actually work when opening notes with `citar-org-roam`:
1. Set `citar-notes-source` to `'citar-org-roam`.
2. Run `citar-open` and select a ref with notes.
3. Select a note and press RET.

It just opens a new file named like `"citekey nodeid"`. This makes sense because `citar-open` calls `citar--open-multi`, which has no idea what to do with a string returned by a note source's `:items` function. It just treats it like a filename and calls `find-file` on it.

Reply:
Ok.

I don't know what happened. Could have sworn it worked on main.

But I'm kind of distracted ATM.

Reply:
OK, I see now. I'm not sure how to fix it ATM though.

Reply:
I see the problem, though not yet a solution:

`citar--select-resource` returns a plain string.

Reply:
We might need an additional note property; a function to say what to do with the string?

But we still have no way of knowing it's a note.